skfp/model_selection/scaffold_split.py

```diff
@@ -0,0 +1,365 @@
+        If train_size, test_size and valid_size aren't set, train_size is set to 0.1.
+
+    include_chirality: bool, default=False
+        Whether to take chirality of molecules into consideration.
+
+    return_indices : bool, default=False
+        Whether the method should return the input object subsets, i.e. SMILES strings
+        or RDKit `Mol` objects, or only the indices of the subsets instead of the data.
+
+    Returns
+    ----------
+    subsets : tuple[list, list, ...]
+    Tuple with train-test subsets of provided arrays. First two are lists of SMILES strings or RDKit `Mol` objects,
+    depending on the input type. If `return_indices` is True, only lists of indices are returned,
+    and any additional data is ignored.
+
+    References
+    ----------
+    .. [1] `Bemis, G. W., & Murcko, M. A.
+        "The properties of known drugs. 1. Molecular frameworks."
+        Journal of Medicinal Chemistry, 39(15), 2887-2893.
+        https://www.researchgate.net/publication/14493474_The_Properties_of_Known_Drugs_1_Molecular_Frameworks`_
+
+    .. [2] `Z. Wu, B. Ramsundar, E. N. Feinberg, J. Gomes, C. Geniesse, A. S. Pappu, K. Leswing, V. Pande
+        "MoleculeNet: A Benchmark for Molecular Machine Learning."
+        Chemical Science, 9(2), 513-530.
+        https://www.researchgate.net/publication/314182452_MoleculeNet_A_Benchmark_for_Molecular_Machine_Learning`_
+
+    """
+    train_size, valid_size, test_size = validate_train_valid_test_split_sizes(
+        train_size, valid_size, test_size
+    )
+
+    scaffolds = _create_scaffolds(list(data), include_chirality)
+    scaffold_sets = sorted(scaffolds.values(), key=len)
+
+    train_ids: list[int] = []
+    valid_ids: list[int] = []
+    test_ids: list[int] = []
+    desired_test_size = int(test_size * len(data))
+    desired_valid_size = int((test_size + valid_size) * len(data))
+
+    for scaffold_set in scaffold_sets:
+        if len(test_ids) < desired_test_size:
+            test_ids.extend(scaffold_set)
+        elif len(valid_ids) < desired_valid_size:
+            valid_ids.extend(scaffold_set)
+        else:
+            train_ids.extend(scaffold_set)
+
+    if not train_ids:
+        raise ValueError(
+            "Train set is empty. Adjust the train_size or check provided data."
+        )
+    if not test_ids:
+        raise ValueError(
+            "Test set is empty. Adjust the test_size or check provided data."
+        )
+
+    train_subset: list[Any] = []
+    valid_subset: list[Any] = []
+    test_subset: list[Any] = []
+
+    if return_indices:
+        train_subset = train_ids
+        valid_subset = valid_ids
+        test_subset = test_ids
+    else:
+        train_subset = get_data_from_indices(data, train_ids)
+        valid_subset = get_data_from_indices(data, valid_ids)
+        test_subset = get_data_from_indices(data, test_ids)
+
+    ensure_nonempty_list(train_subset)
+    ensure_nonempty_list(valid_subset)
```

Comment:
I think this shoule be `test_subset`. Otherwise warning below doesn't really make sense

Reply:
Correct